src/components/Accordion.js

```diff
@@ -51,47 +51,50 @@ if (isIE) {
  */
 const toggleAccordionFocus = focusEvent => {
   const { target, type: focusType } = focusEvent;
-  var accordionHeaderTextElms = target.getElementsByClassName(
+  var accordionHeaderElms = target.getElementsByClassName(
     "dg_accordion_buttontext-holder"
   );
 
   if (focusType === "focusin") {
-    accordionHeaderTextElms[0].classList.add("ms-focus-within");
+    accordionHeaderElms[0].classList.add("ms-focus-within");
   }
 
   if (focusType === "focusout") {
-    accordionHeaderTextElms[0].classList.remove("ms-focus-within");
+    accordionHeaderElms[0].classList.remove("ms-focus-within");
   }
 };
 
-const selectElementByClassName = (element, cssName) => {
+const selectElementByClassName = (element, cssNameText) => {
   var sibling = element.nextElementSibling;
   while (sibling) {
-    if (sibling.matches(cssName)) return sibling;
-    sibling = sibling.nextElementSibling;
+    if (sibling.matches(cssNameText)) {
+      return sibling;
+    } else {
+      sibling = sibling.nextElementSibling;
+    }
   }
 };
 
 const menuAction = accordionHeaderElm => {
-  var mainDiv = accordionHeaderElm.closest(".dg_accordion");
-  var menuItems = mainDiv.getElementsByClassName("multi-collapse");
-  var totalCollapsed = mainDiv.getElementsByClassName("show");
-  var accordionButton = accordionHeaderElm.closest("button");
-  var accordionContent = selectElementByClassName(
-    accordionButton,
+  var mainDivElm = accordionHeaderElm.closest(".dg_accordion");
+  var menuItems = mainDivElm.getElementsByClassName("multi-collapse");
+  var totalCollapsedPanels = mainDivElm.getElementsByClassName("show");
+  var accordionButtonElm = accordionHeaderElm.closest("button");
+  var accordionContentElm = selectElementByClassName(
```

Comment:
not blcoking since this is working, but is there a reason you didn't use getElementsByClassName here?

Reply:
So in this case closest didn't work since the element I was looking for is below the parent and I believe closest only goes up the nodes. I didn't use getElementsByClassName because it returns an "array" and I only needed the one element. I kind of don't like using Element[0] since if at any point the array changes (new element is added or something like that) then it might break. That argument can be used for other sections as well though I guess. Anyway the solution was to grab a node and loop through to find the first element with that class and then return that. I had planned on putting that function into a utility class to be used in other locations but haven't found the time yet.

Reply:
cool thanks for explaining, one way to handle this is to take your target and immediately find the closest parent aka accordion. Then from that variable, you can find the specific parts that you need.